_entries/02-03 challenge2-2.md

```diff
@@ -18,20 +18,29 @@ In the table below, you will find the Docker container images provided by the de
 
 ### Environment variables
 
-The Order Capture API requires certain environment variables to properly run and track your progress. Make sure you set those environment variables.
+The Order Capture API requires certain environment variables to properly run and track your progress. Make sure you set those environment variables in your deployment (you don't need to set these locally in your shell).
 
   * `MONGOHOST="<hostname of mongodb>"`
-    * MongoDB hostname. Read from a Kubernetes secret called **mongodb**.
+    * MongoDB hostname. Read directly as a string value in the YAML.
```

Comment:
Have you tested the DevOps pipeline with this change?

Reply:
Specifically, with this change, this line will not work
https://github.com/Azure/azch-captureorder/blob/master/azure-pipelines.yml#L69

Reply:
I have not tested the pipeline, I wasn't aware of it

Reply:
Can we keep it as a secret? You can argue that a username as well doesn't need to be stored in a secret, however the [docs](https://kubernetes.io/docs/concepts/configuration/secret/) show otherwise.

Reply:
I'm happy to revert this change, I don't feel super strongly about it
But it does seem weird storing a simple hostname as a secret

I've tested it with this in my deployment and it worked fine
```
          env:
          - name: MONGOHOST
            value: "orders-mongo-mongodb.default.svc.cluster.local"
          - name: MONGOUSER
            valueFrom:
              secretKeyRef:
                name: mongodb
                key: mongoUser
          - name: MONGOPASSWORD
            valueFrom:
              secretKeyRef:
                name: mongodb
                key: mongoPassword
```

Reply:
OK, I'll switch it back in the morning

Reply:
Switched it back so that MONGOHOST/mongoHost is a secret again. Added a little explanation on what to store in the secret and where mongoHost gets it's value from. I've seen people very confused about why it is `orders-mongo-mongodb.default.svc.cluster.local`